### PR TITLE
Add test for header error conditions

### PR DESCRIPTION
**Pull request by acolytec3**
Partially addresses #1232 - Add 3 new tests for error conditions in `header.ts` that are currently not covered.

### CODE REVIEW

**Comment by acolytec3** [comment]
@holgerd77 As discussed in email, this is my first pass at a small to add coverage for part of `header.ts` that is currently missed in test coverage.

File: packages/block/test/header.spec.ts [hunk 1]
```diff
@@ -280,6 +280,20 @@ tape('[Block]: Header functions', function (t) {
     }
     headerData.mixHash = Buffer.alloc(32)
 
+    testCase = 'show throw on invalid clique difficulty'
```
Comment (holgerd77) [inline packages/block/test/header.spec.ts:283]: show -> should

File: packages/block/test/header.spec.ts [hunk 2]
```diff
@@ -344,4 +358,35 @@ tape('[Block]: Header functions', function (t) {
     )
     st.end()
   })
+
+  t.test('should throw on fromRLPSerializedHeader() with header as rlp encoded string', function (st) {
+    const badHeader = function (): void {
+      let header = BlockHeader.fromRLPSerializedHeader(rlp.encode('a'))
+    }
+    st.throws(() => badHeader(), 'Invalid serialized header input. Must be array')
```
Comment (holgerd77) [inline packages/block/test/header.spec.ts:366]: Some nit: "Invalid" -> "invalid" (all other test comments start with lowercase)

File: packages/block/test/header.spec.ts [hunk 3]
```diff
@@ -344,4 +358,35 @@ tape('[Block]: Header functions', function (t) {
...
+    st.end();
+  })
+
+  t.test('should throw on fromValuesBuffer() call with values array with length > 15', function (st) {
+    const badHeader = function (): void {
+      const zero = Buffer.alloc(0)
+      const headerArray = []
+      for (let item = 0; item < 16; item++) {
+        headerArray.push(zero)
+      }
```
Comment (holgerd77) [inline packages/block/test/header.spec.ts:376]: You can simplify with: `Array(16).fill(Buffer.alloc(0))`

**Review by holgerd77 — CHANGES_REQUESTED** [review]
Hi @acolytec3,
nice, thanks a lot for this! 😄 

Some minor remarks, also linting fails on CI. Can you please run `npm run lint:fix`, and fix non-fixable things manually afterwards and commit and push? This is for having somewhat better code consistency and is always checked along a CI run.

**Comment by acolytec3** [comment]
@holgerd77 I've added functional tests for the `validateCliqueDifficulty` method in `header.ts` and I believe addressed all the linting issues.

**Review by holgerd77 — APPROVED** [review]
Thanks a lot, like this clean Mockchain setup 😄 , LGTM!

**Comment by holgerd77** [comment]
Ah, maybe some additional nit: since we are in a monorepo please label your both your PR title (e.g. "Block: Add...", have done this now directly) and your commits (e.g. `block: add test for header error conditions`) with the libraries touched, otherwise this is not giving enough context.

**Pull request merged**